08_flask_html_basics.md

```diff
@@ -0,0 +1,592 @@
+
+- It will logically separate and aggregate content that is related
+- It can allow easier styling using a language like CSS - this is beyond the scope of this exercise
+
+Similarly, for text-based content, you could use the paragraph [`<p>`](https://www.w3schools.com/tags/tag_p.asp) tag.
+
+That is all the HTML that we will need for now, but we will come back to it later, to talk about forms, inputs and buttons.
+
+## Web Servers
+
+> Web servers are applications and hardware that are responsible for delivering content available at a certain URL, when a client makes a HTTP request over the internet.
+
+Both the application responsible for delivering content, the hardware on which it runs, or the system as a whole are called "web server". It's usually not important to explicitly differentiate between them, but in this exercise, "web server" will refer to the application that you write.
+
+Let's think of what happens when you access a site over the internet:
+
+- You navigate to a URL in your browser
+- This will generate a HTTP GET request (more on this later!)
+- The request will be received by the web server (hardware & software) associated with the domain name (e.g. google.com)
+- The web server will check whether the resource that is requested is mapped to a certain route. In this context, a _route_ is the path after the `<hostname>/` on the URL. Looking at the google search example (`https://www.google.com/search?q=alpaca/`), the resource is mapped to the `/search` route.
+- If the resource exists, the server responds with a GET Response, which can have many forms. For now, let's say it's an HTML page.
+- The browser receives the response and displays it. If it's an HTML page, it will render the HTML.
+
+The current HTML document is only accessible from your filesystem. Ideally, however, the page to be accessible from the internet, or at least from your machine at a specific URL.
+
+The next goal is to use an existing web server, make it run locally, and customize it in Python to serve your page when accessing its URL in the browser.
+
+## Flask
+
+You can use the [Flask](https://flask.palletsprojects.com/en/2.2.x/) framework to help with the goals above.
+
+Flask is a framework that provides a built-in web server, as well as a library that allows you to create a web application.
+
+Let's create a new folder called `zoo_app` and open it in VS Code.
+
+### Exercise 8.7: Adding dependencies
+
+As in previous exercises, you will use `poetry` to manage python packages
+- Run the command to initialize poetry, `poetry init`. You can use default settings
+- In order to use a virtual environment local to your project, you will need to create a new file called `poetry.toml` in the root of your project, and add the following two lines to it:
+```python
+[virtualenvs]
+create = true
+in-project = true
+```
+- Run the command to add `flask` to the list of dependencies
```

Comment:
I intentionally wanted to leave it blank so people would need to search for it/remember it, but I can see now that this might be confusing. I'll just add the command